### ISSUE TITLE
Backprop test required not only for loss op

### ISSUE THREAD

**Issue by zchen0211** [PaddlePaddle/Paddle#3541]
In our python tests, we only have backward-propagation tests for loss ops. (e.g., the cross entropy op).

Actually, we also need them for common op, we also need them (e.g., add op).
We need to input a gradient from output and bp it back and check the correctness.

**Comment by jacquesqiao** [comment]
the auto gradient check is for common op now, https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/v2/framework/tests/test_softmax_op.py#L27,  some op have not add test now, but the test if ready for all of them.

**Comment by typhoonzero** [comment]
Closing due to low activity, please feel free to reopen it.
